planetscale/regions.go

```diff
@@ -0,0 +1,49 @@
+package planetscale
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+const regionsAPIPath = "v1/regions"
+
+type Region struct {
+	Slug string `json:"slug"`
+	Name string `json:"display_name"`
+}
+
+type regionsResponse struct {
+	Regions []*Region `json:"data"`
+}
+
+type RegionsService interface {
+	List(ctx context.Context) ([]*Region, error)
```

Comment:
I think we should use a function signature like this:
```
	List(context.Context, *ListRegionRequest) ([]*Region, error)
```

That way, in the future, if we decide to provide the option to list only specific regions, it'll not break existing applications.

Reply:
Fixed in https://github.com/planetscale/planetscale-go/pull/69/commits/022bba71fcf64ebcfa798b100c95ba2ea77975bb